bin/generate_npm_tarball.sh

```diff
@@ -10,12 +10,11 @@ wd=$(mktemp -d)
 trap "rm -r '$wd'" EXIT INT TERM
 
 create_cache() {
-  node_modules/npm/bin/npm-cli.js install --cache $wd/cache $package --no-shrinkwrap --no-optional --production --verbose
+  npm install --cache $wd/cache $package --no-shrinkwrap --no-optional --production --verbose
 }
 
 mkdir $wd/cache $wd/install
 cd $wd/install
-npm install npm@3.x
```

Comment:
I dropped this to force defaulting to the node/npm used when generating since the tarball is dependent on the target version of nodejs.

Reply:
It's mostly a leftover from when we built with npm 2 and npm 3. The version does matter because NPM might cache in a different way.

Reply:
One thing I did notice is that this tarball includes the actual source files so we end up storing them twice. Might be good to address that as well.

Reply:
Can you expand a bit more? I am running v10.13.0 and the only thing the tarball has in its the cache with no source files.

Reply:
Perhaps that was a problem with NPM 3. If you open `packages/foreman/nodejs-webpack/webpack-3.12.0-registry.npmjs.org.tgz` you will see it actually includes the `package.tgz` as well even though we add that as a `SourceNN` which means the tarball is much larger. Since it's stored in git, it actually blows up our entire repo.

Reply:
When I run this script I run into this issue:

```
rm: remove write-protected regular file '/tmp/tmp.zYRq5nlXrN/cache/_cacache/content-v2/sha512/94/13/7dccb37fa007faf28df335762b742b7590ff16b22196f0ea3691ae356f620ce492ff4b5093c97d6b5b499bff34ae26e9f4654ac3c71e2caaf612d855b33c'? 
```

Any ideas?

Reply:
The generated tarball has no `package.tgz` and ends up (testing against nodejs-history) being about 35K smaller.